### PR TITLE
Implement SA1132 + unit tests

### LINKED ISSUE #825: Implement SA1132 (DoNotCombineFields)
**Category:** Readability
**Title:** Do not combine field definitions
**Description:** C# allows more than one field to be defined in a single field declaration syntax. When fields are combined, it becomes difficult to impossible to properly document them, and difficult to identify each field defined in a type.

**Example:**

The following code shows an example which would produce this warning:

```csharp
int x, y;
```

To resolve the violation, define in field separately:

```csharp
int x;
int y;
```

### PR DESCRIPTION
**Pull request by Maxwe11**
Fixes #825

### CODE REVIEW

**Comment by vweijsters** [comment]
:+1: Looks good to me

**Comment by Maxwe11** [comment]
@vweijsters could you review second commit please? :smile:

**Comment by vweijsters** [comment]
:+1:

**Comment by vweijsters** [comment]
The current code fix code will duplicate any trivia that are not part of the variable declarations. For example:
```C#
public class TestClass
{
#if DEBUG
    private int field1, field2;
    private int field3;
#endif
}
```
will be fixed to:
```C#
public class TestClass
{
#if DEBUG
    private int field1;
#if DEBUG
    private int field2;
    private int field3;
#endif
}
```

**Comment by Maxwe11** [comment]
@vweijsters see 89e77c4 for a fix. Previous commits just squash as is.

File: StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1132CodeFixProvider.cs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+            if (fieldDeclaration != null)
+            {
+                VariableDeclarationSyntax declaration = fieldDeclaration.Declaration;
+                SeparatedSyntaxList<VariableDeclaratorSyntax> variables = declaration.Variables;
+                VariableDeclaratorSyntax first = variables.First();
+                var newFieldDeclarations = new List<BaseFieldDeclarationSyntax>(variables.Count);
+
+                foreach (VariableDeclaratorSyntax variable in variables)
+                {
+                    var variableDeclarator = SyntaxFactory.SeparatedList(new[] { variable });
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1132CodeFixProvider.cs]: :bulb: The downside of this syntax is it forces the code to use this overload even if a future release includes an overload of `SeparatedList` that takes exactly one argument.

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1132UnitTests.cs [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        [InlineData("public event System.Action a;")]
+        public async Task TestValidDeclarationAsync(string declaration)
+        {
+            var testCode = GetTemplate(declaration);
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Theory]
+        [InlineData("private int a, b;", "private int a;\r\n    private int b;")]
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1132UnitTests.cs]: :thought_balloon: I find this hard to read. I think I would have preferred two separate test methods instead.

**Comment by vweijsters** [comment]
:+1: The fix for directives is OK. The code fix will still duplicate leading comments / trialing comments, but that is purely cosmetic and can be fixed later if it annoys people.

File: StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1132CodeFixProvider.cs (hunk 1 above)
Comment (Maxwe11) [inline StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1132CodeFixProvider.cs]: fixed

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1132UnitTests.cs (hunk 2 above)
Comment (Maxwe11) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1132UnitTests.cs]: fixed

**Pull request merged**